src/pretix/helpers/payment.py

```diff
@@ -0,0 +1,208 @@
+# applicable granting you additional permissions and placing additional restrictions on your usage of this software.
+# Please refer to the pretix LICENSE file to obtain the full terms applicable to this work. If you did not receive
+# this file, see <https://pretix.eu/about/en/license>.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License along with this program.  If not, see
+# <https://www.gnu.org/licenses/>.
+#
+from urllib.parse import quote, urlencode
+
+import text_unidecode
+from django.utils.safestring import mark_safe
+from django.utils.translation import gettext_lazy as _
+
+
+def dotdecimal(value):
+    return str(value).replace(",", ".")
+
+
+def commadecimal(value):
+    return str(value).replace(".", ",")
+
+
+def generate_payment_qr_codes(
+        event,
+        code,
+        amount,
+        bank_details_sepa_bic,
+        bank_details_sepa_name,
+        bank_details_sepa_iban,
+):
+    out = []
+    for method in [
+        swiss_qrbill,
+        czech_spayd,
+        euro_epc_qr,
+        euro_bezahlcode,
+    ]:
+        data = method(
+            event,
+            code,
+            amount,
+            bank_details_sepa_bic,
+            bank_details_sepa_name,
+            bank_details_sepa_iban,
+        )
+        if data:
+            out.append(data)
+
+    return out
+
+
+def euro_epc_qr(
+        event,
+        code,
+        amount,
+        bank_details_sepa_bic,
+        bank_details_sepa_name,
+        bank_details_sepa_iban,
+):
+    if event.currency != 'EUR' or not bank_details_sepa_iban:
+        return
+
+    return {
+        "id": "girocode",
+        "label": "EPC-QR",
+        "qr_data": "\n".join(text_unidecode.unidecode(str(d or '')) for d in [
+            "BCD",   # Service Tag: ‘BCD’
+            "002",   # Version: V2
+            "2",     # Character set: ISO 8859-1
+            "SCT",   # Identification code: ‘SCT‘
+            bank_details_sepa_bic,   # AT-23 BIC of the Beneficiary Bank
+            bank_details_sepa_name,  # AT-21 Name of the Beneficiary
+            bank_details_sepa_iban,  # AT-20 Account number of the Beneficiary
+            f"{event.currency}{dotdecimal(amount)}",  # AT-04 Amount of the Credit Transfer in Euro
+            "",      # AT-44 Purpose of the Credit Transfer
+            "",      # AT-05 Remittance Information (Structured)
+            code,    # AT-05 Remittance Information (Unstructured)
+            "",      # Beneficiary to originator information
+            "",
+        ]),
+    }
+
+
+def euro_bezahlcode(
+        event,
+        code,
+        amount,
+        bank_details_sepa_bic,
+        bank_details_sepa_name,
+        bank_details_sepa_iban,
+):
```

Comment:
Not really part of this PR, but while we're at it, we could probably limit this to IBANs starting with DE. It's a very German thing and more or less on its way out.